dc/s2n-quic-dc/src/stream/server/tokio/tcp/manager.rs

```diff
@@ -317,18 +318,13 @@ where
 
     #[cfg(debug_assertions)]
     fn invariants(&self) {
-        for idx in 0..self.workers.len() {
-            assert!(
-                self.free
-                    .iter(&self.workers)
-                    .chain(self.by_sojourn_time.iter(&self.workers))
-                    .filter(|v| *v == idx)
-                    .count()
-                    == 1,
-                "worker {idx} should be linked at all times\n{:?}",
-                self.workers[idx].link,
-            );
-        }
+        let mut linked_workers = self
+            .free
+            .iter(&self.workers)
+            .chain(self.by_sojourn_time.iter(&self.workers))
+            .collect::<Vec<_>>();
+        linked_workers.sort();
+        assert!((0..self.workers.len()).eq(linked_workers.iter().copied()));
```

Comment:
We probably need an assert! that the linked_workers has the same length? I guess sort of implies with free.iter, but maybe doesn't hurt?

Reply:
`Iterator::eq` should also check the length: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.eq

```rust
assert_eq!([1].iter().eq([1].iter()), true);
assert_eq!([1].iter().eq([1, 2].iter()), false);
```

Reply:
Hm, I'm distinctly remembering some kind of oops around this... maybe just bad memory.

Reply:
I had the same memory at first. There are some that so have that issue, like `zip`.